pkgs/development/compilers/dotnet/default.nix

```diff
@@ -89,39 +90,82 @@ let
       }
     );
   };
+
+  # combine an SDK with the runtime/packages from a base SDK
+  combineSdk =
+    base: overlay:
+    if (overlay.runtime.version != base.runtime.version) then
+      throw "combineSdk: unable to combine ${overlay.name} with ${base.name} because runtime versions don't match (${overlay.runtime.version} != ${base.runtime.version})"
+    else
+      pkgs.callPackage ./wrapper.nix { } "sdk" (
+        (pkgs.combinePackages [
+          base.runtime
+          base.aspnetcore
+          (overlay.overrideAttrs (old: {
+            passthru = old.passthru // {
+              inherit (base)
+                packages
+                targetPackages
+                ;
+            };
+          }))
+        ]).unwrapped.overrideAttrs
+          (old: {
+            name = overlay.unwrapped.name;
+            # resolve symlinks so DOTNET_ROOT is self-contained
+            postBuild = ''
+              mv "$out"/share/dotnet{,~}
```

Comment:
I did this for now to fix the F# problem and to avoid pulling both dependencies into the closure.

I didn't want to change the behaviour or interface of `combinePackages` just yet.